ReactiveCocoa/NSObject+Lifetime.swift

```diff
@@ -10,6 +10,27 @@ fileprivate let lifetimeKey = AssociationKey<Lifetime?>(default: nil)
 /// Holds the `Lifetime.Token` of the object.
 fileprivate let lifetimeTokenKey = AssociationKey<Lifetime.Token?>(default: nil)
 
+internal func lifetime(of object: AnyObject) -> Lifetime {
```

Comment:
Maybe we should expose this in the future. And maybe even create a shortcut version of `take(during: lifetime(of: object))`. 🤔